web_ui/src/core/jobs/hooks/utils.test.tsx

```diff
@@ -36,32 +36,39 @@ describe('Use jobs hook utils', () => {
     });
 
     it('Should not invalidate balance if feature flag is disabled', async () => {
-        const queryClient = new QueryClient();
-        queryClient.invalidateQueries = jest.fn();
+        const queryClient = renderHookWithProviders(useQueryClient);
+        queryClient.result.current.invalidateQueries = jest.fn();
+
         renderHookWithProviders(
-            () =>
-                useInvalidateBalanceOnNewJob(
+            () => {
+                return useInvalidateBalanceOnNewJob(
                     workspaceIdentifier,
                     getMockedResponse([getMockedJob({ cost: { leaseId: '123', requests: [], consumed: [] } })]),
                     { jobState: JobState.SCHEDULED }
-                ),
+                );
+            },
             {
-                providerProps: { queryClient, featureFlags: { FEATURE_FLAG_CREDIT_SYSTEM: false } },
+                providerProps: { featureFlags: { FEATURE_FLAG_CREDIT_SYSTEM: false } },
             }
         );
 
         await waitFor(() => {
-            expect(queryClient.invalidateQueries).not.toHaveBeenCalled();
+            expect(queryClient.result.current.invalidateQueries).not.toHaveBeenCalled();
         });
     });
 
     it('Should invalidate balance if feature flag is enabled', async () => {
-        const queryClient = new QueryClient();
-        queryClient.invalidateQueries = jest.fn();
+        let invalidateSpy: jest.SpyInstance | undefined;
```

Comment:
Discussed offline about these. I just "refactored" the tests to make it work with the new approach, but fundamentally they are not great tests in the sense that they test if react-query works instead of testing our app behavior AFTER react-query acts. So we should check if , once we do X, then Y happens, not that react-query called its internals correctly.